lib/widgets/message_list.dart

```diff
@@ -1445,9 +1458,67 @@ class MessageWithPossibleSender extends StatelessWidget {
         child: Icon(ZulipIcons.star_filled, size: 16, color: designVariables.star));
     }
 
+    final editMessageErrorStatus = store.getEditMessageErrorStatus(message.id);
+    Widget? editMessageErrorStatusRow;
+    if (editMessageErrorStatus != null) {
+      final baseTextStyle = TextStyle(
+        fontSize: 12,
+        height: 12 / 12,
+        letterSpacing: proportionalLetterSpacing(context,
+          0.05, baseFontSize: 12),
+      );
+      editMessageErrorStatusRow = switch (editMessageErrorStatus) {
+        // TODO parse markdown and show new content as local echo?
+        false => Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          spacing: 1.5,
+          children: [
+            Text(
+              style: baseTextStyle.copyWith(color: designVariables.btnLabelAttLowIntInfo),
+              textAlign: TextAlign.end,
+              zulipLocalizations.savingMessageEditLabel),
+            // TODO instead place within outer padding; see Figma:
+            //   https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=4026-8775&m=dev
+            LinearProgressIndicator(
+              minHeight: 2,
+              color: designVariables.foreground.withValues(alpha: 0.5),
+              backgroundColor: designVariables.foreground.withValues(alpha: 0.2),
+            ),
+          ]),
+        true => _withRestoreEditMessageGestureDetector(context,
+          Text(
+            style: baseTextStyle.copyWith(color: designVariables.btnLabelAttLowIntDanger),
+            textAlign: TextAlign.end,
+            zulipLocalizations.savingMessageEditFailedLabel)),
+      };
+    }
+
+    Widget content = MessageContent(message: message, content: item.content);
+
+    if (editMessageErrorStatus != null) {
+      content = Opacity(opacity: 0.6, child: content);
+      switch (editMessageErrorStatus) {
+        case true:
+          content = _withRestoreEditMessageGestureDetector(context, content);
```

Comment:
This also seems like a good candidate for its own widget:
```suggestion
          content = _RestoreEditMessageGestureDetector(messageId: item.message.id, child: content);
```